File: admin-service/src/main/java/com/algoforge/adminservice/service/AdminService.java
package com.algoforge.adminservice.service;

import com.algoforge.common.dto.AlgoUserDto;
import com.algoforge.common.feign.AuthServiceClient;
import com.algoforge.common.feign.ContestServiceClient;
import com.algoforge.common.feign.TaskServiceClient;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AdminService {

    @Autowired
    private AuthServiceClient authServiceClient;

    @Autowired
    private TaskServiceClient taskServiceClient;

    @Autowired
    private ContestServiceClient contestServiceClient;

    public AlgoUserDto blockUser(String username) {
        return authServiceClient.blockUser(username);
    }

    public AlgoUserDto unblockUser(String username) {
        return authServiceClient.unblockUser(username);
    }

    public void deleteUser(String username) {
        authServiceClient.deleteUser(username);
    }

    public void deleteTask(Long taskId) {
        taskServiceClient.deleteTask(taskId);
    }

    public void deleteContest(Long contestId) {
        contestServiceClient.deleteContest(contestId);
    }

}

File: front/algoforge/src/app/core/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface AlgoUserDto {
  id: number;
  email: string;
  username: string;
  bio?: string;
  profilePhoto?: string; // base64 или URL
  creationDate: string; // ISO string
  rating: number;
  isBlocked: boolean;
  isDeleted: boolean;
  roles: string[];
}

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  private apiUrl = 'http://localhost:8080/api/admin';

  constructor(private http: HttpClient) { }

  blockUser(username: string): Observable<AlgoUserDto> {
    return this.http.put<AlgoUserDto>(`${this.apiUrl}/users/${username}/block`, null);
  }

  unblockUser(username: string): Observable<AlgoUserDto> {
    return this.http.put<AlgoUserDto>(`${this.apiUrl}/users/${username}/unblock`, null);
  }

  deleteUser(username: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/users/${username}`);
  }

  deleteTask(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/tasks/${id}`);
  }

  deleteContest(contestId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/contests/${contestId}`);
  }
}
